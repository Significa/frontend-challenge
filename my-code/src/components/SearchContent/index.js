import React from 'react'
import PropTypes from 'prop-types'

import SearchResults from '../../containers/SearchResults'
import Message from '../Message'

export default function SearchContent({
  search,
  results,
  error,
  errorMessage
}) {
  if (error) {
    return <Message title="Something went wrong" subtitle={errorMessage} />
  }

  if (!search) {
    return (
      <Message
        title="Don’t know what to search?"
        subtitle="Here’s an offer you can’t refuse"
      />
    )
  }

  if (!results) {
    return (
      <Message
        title="Nothing found"
        subtitle="Try searching for something else"
      />
    )
  }

  return <SearchResults />
}

SearchContent.propTypes = {
  search: PropTypes.bool.isRequired,
  results: PropTypes.bool.isRequired,
  error: PropTypes.bool.isRequired,
  errorMessage: PropTypes.string
}

SearchContent.defaultProps = {
  errorMessage: undefined
}
