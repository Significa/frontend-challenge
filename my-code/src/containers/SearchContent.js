import { connect } from 'react-redux'

import getQueryParam from '../utils/get-query-param'
import {
  getIsFailed,
  getIsLoading,
  getErrorMessage
} from '../state/modules/api-status'

import SearchContent from '../components/SearchContent'

const mapStateToProps = ({ router, searchResults, apiStatus }) => ({
  search: !!getQueryParam(router.location.search, 'search'),
  results: !!searchResults.length && !getIsLoading(apiStatus),
  error: getIsFailed(apiStatus),
  errorMessage: getErrorMessage(apiStatus)
})

export default connect(mapStateToProps)(SearchContent)
