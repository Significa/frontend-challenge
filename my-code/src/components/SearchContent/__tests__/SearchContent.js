import React from 'react'
import { shallow } from 'enzyme'

import SearchContent from '../'

describe('SearchContent', () => {
  it('renders correctly', () => {
    const tree = shallow(<SearchContent error={false} search results />)
    expect(tree).toMatchSnapshot()
  })

  it('renders error correctly', () => {
    const tree = shallow(
      <SearchContent
        error
        search={false}
        results={false}
        errorMessage="Error message goes here"
      />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders no search correctly', () => {
    const tree = shallow(
      <SearchContent error={false} search={false} results={false} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders no results correctly', () => {
    const tree = shallow(<SearchContent error={false} search results={false} />)
    expect(tree).toMatchSnapshot()
  })
})
