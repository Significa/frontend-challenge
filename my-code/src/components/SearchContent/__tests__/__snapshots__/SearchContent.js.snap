// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContent renders correctly 1`] = `<Connect(SearchResults) />`;

exports[`SearchContent renders error correctly 1`] = `
<Message
  subtitle="Error message goes here"
  title="Something went wrong"
/>
`;

exports[`SearchContent renders no results correctly 1`] = `
<Message
  subtitle="Try searching for something else"
  title="Nothing found"
/>
`;

exports[`SearchContent renders no search correctly 1`] = `
<Message
  subtitle="Here’s an offer you can’t refuse"
  title="Don’t know what to search?"
/>
`;
